dbcon/joblist/jlf_subquery.cpp

```diff
@@ -90,19 +90,32 @@ void getColumnValue(ConstantColumn** cc, uint64_t i, const Row& row, const strin
 
         case CalpontSystemCatalog::DECIMAL:
         case CalpontSystemCatalog::UDECIMAL:
-            data = row.getIntField(i);
-            oss << (data / IDB_pow[row.getScale(i)]);
-
-            if (row.getScale(i) > 0)
+            if (row.getColumnWidth(i) == datatypes::MAXDECIMALWIDTH)
             {
-                if (data > 0)
-                    oss << "." << (data % IDB_pow[row.getScale(i)]);
-                else if (data < 0)
-                    oss << "." << (-data % IDB_pow[row.getScale(i)]);
+                int128_t val;
+                row.getInt128Field(i, val);
+                IDB_Decimal dec(0, row.getScale(i), row.getPrecision(i), val);
```

Comment:
I will remove the `row.getInt128Field()` in favour of `row.getTSInt128Field()` once the new IDB_Decimal ctor taking a `TSInt128` as an argument is available as part of https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/1610

Reply:
You can even use a ctor with 3 arguments that doesn't set the int64 value in the Decimal